tests/vpn_service_test.py

```diff
@@ -0,0 +1,176 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for vpn service"""
+import pickle
+import sys
+
+from perfkitbenchmarker import benchmark_spec
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import linux_benchmarks
+from perfkitbenchmarker import providers
+from perfkitbenchmarker.configs import benchmark_config_spec
+from perfkitbenchmarker.vpn_service import TunnelConfig
+from tests import pkb_common_test_case
+
+FLAGS = flags.FLAGS
+
+PROJECT = 'mock_project'
+CLOUD = providers.GCP
+BENCHMARK_NAME = 'iperf'
+URI = 'uri45678'
+
+DEFAULT_CFG = """
+# VPN iperf config.
+iperf:
+  description: Run iperf over vpn
+  flags:
+    iperf_sending_thread_count: 5
+    use_vpn: True
+    vpn_service_gateway_count: 1
+  vpn_service:
+    tunnel_count: 1
+    ike_version: 2
+    routing_type: static
+  vm_groups:
+    vm_1:
+      cloud: GCP
+      cidr: 10.0.1.0/24
+      vm_spec:
+        GCP:
+            zone: us-west1-b
+            machine_type: n1-standard-4
+    vm_2:
+      cloud: GCP
+      cidr: 192.168.1.0/24
+      vm_spec:
+        GCP:
+            zone: us-central1-c
+            machine_type: n1-standard-4
+"""
+
+
+class BaseVPNServiceTest(pkb_common_test_case.PkbCommonTestCase):
+
+  def setUp(self):
+    super(BaseVPNServiceTest, self).setUp()
+    if not sys.warnoptions:  # https://bugs.python.org/issue33154
+      import warnings
+      warnings.simplefilter("ignore", ResourceWarning)
+
+  def _CreateBenchmarkSpecFromYaml(self, yaml_string,
+                                   benchmark_name=BENCHMARK_NAME):
+    config = configs.LoadConfig(yaml_string, {}, benchmark_name)
+    return self._CreateBenchmarkSpecFromConfigDict(config, benchmark_name)
+
+  def _CreateBenchmarkSpecFromConfigDict(self, config_dict, benchmark_name):
+    config_spec = benchmark_config_spec.BenchmarkConfigSpec(benchmark_name,
+                                                            flag_values=FLAGS,
+                                                            **config_dict)
+    benchmark_module = next((b for b in linux_benchmarks.BENCHMARKS if
+                             b.BENCHMARK_NAME == benchmark_name))
+    return benchmark_spec.BenchmarkSpec(benchmark_module, config_spec, URI)
+
+  def extractDictAFromB(self, A, B):  # assertDictContainsSubset deprecated
+    return dict([(k, B[k]) for k in A.keys() if k in B.keys()])
+
+
+class VpnServiceTestCase(BaseVPNServiceTest):
+
+  def testVpnServiceConfig(self):
+    FLAGS.use_vpn = True
```

Comment:
use flagsaver to make tests more robust to order of execution

Reply:
Fixed.